apps/web/src/state/settings/index.ts

```diff
@@ -0,0 +1,28 @@
+import { modules } from '@walless/ioc';
+import { type SettingDocument } from '@walless/store';
+
+import { settingState } from './internal';
+
+export const settingsActions = {
+	setSettings: (settings: SettingDocument) => {
+		if (settings._id) {
+			settingState._id = settings._id;
+			settingState.hideBalance = settings.hideBalance;
+		}
+	},
+	updateHiddenBalanceSettings: async ({
+		_id,
+		hideBalance,
+	}: {
+		_id: string;
+		hideBalance: boolean;
+	}) => {
+		await modules.storage.upsert<SettingDocument>(_id, async (doc) => {
+			doc.hideBalance = hideBalance;
+
+			return doc;
+		});
```

Comment:
once the user signed-in this setting Document must be ready there, no `upsert` is or `put` needed, I think we don't need this method at all.

Reply:
the requirement is when user open extension again, they see the same private setting as the last time they closed extension. Therefore, I think we need a method to update the setting Document, am I right?

Reply:
oops, yes! you're right, I forgot that `upsert` is for update as well.